src/error.rs

```diff
@@ -1,64 +1,81 @@
 use base64::DecodeError;
-use std::{num::ParseIntError, str};
+use reqwest::Error as ReqwestError;
+use serde_json::Error as JsonError;
+use std::io::Error as IoError;
+use std::num::ParseIntError;
+use std::str::Utf8Error;
 use thiserror::Error;
 use websocket::{client::ParseError, WebSocketError};
 
 /// Enumeration of all possible errors in the `socket.io` context.
 #[derive(Error, Debug)]
 #[non_exhaustive]
 pub enum Error {
+    // Conform to https://rust-lang.github.io/api-guidelines/naming.html#names-use-a-consistent-word-order-c-word-order
+    // Negative verb-object
     #[error("Invalid packet id: {0}")]
     InvalidPacketId(u8),
     #[error("Error while parsing an empty packet")]
-    EmptyPacket,
+    EmptyPacket(),
     #[error("Error while parsing an incomplete packet")]
-    IncompletePacket,
+    IncompletePacket(),
     #[error("Got an invalid packet which did not follow the protocol format")]
-    InvalidPacket,
+    InvalidPacket(),
     #[error("An error occurred while decoding the utf-8 text: {0}")]
-    Utf8Error(#[from] str::Utf8Error),
+    InvalidUtf8(#[from] Utf8Error),
     #[error("An error occurred while encoding/decoding base64: {0}")]
-    Base64Error(#[from] DecodeError),
+    InvalidBase64(#[from] DecodeError),
     #[error("Invalid Url: {0}")]
     InvalidUrl(String),
     #[error("Error during connection via http: {0}")]
-    ReqwestError(#[from] reqwest::Error),
+    IncompleteReqwest(#[from] ReqwestError),
```

Comment:
Mhh I don't really like that name because this error captures every error possibly emitted by the `reqwest` crate. But the error name sounds like an error during connection build up, which is not necessarily the case. Maybe something like `FromReqwestError` would be better?

Reply:
Fair point. Looking at the Request library their error type covers everything from redirection, connections, timeouts and more. perhaps:
```suggestion
IncompleteResponceFromReqwest
```
Trying to keep the error types constructed in the same form: (negative verb-object(-description of state)

Reply:
Well that is true. I see the point and I like your suggestion. Lets go with `IncompleteResponseFromReqwest`.